aws/src/main/java/org/apache/iceberg/aws/s3/S3AccessGrantsPluginConfigurations.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws.s3;
+
+import java.util.Map;
+import org.apache.iceberg.util.PropertyUtil;
+import software.amazon.awssdk.s3accessgrants.plugin.S3AccessGrantsPlugin;
+import software.amazon.awssdk.services.s3.S3ClientBuilder;
+
+class S3AccessGrantsPluginConfigurations {
+  private boolean isS3AccessGrantsFallbackToIAMEnabled;
+
+  private S3AccessGrantsPluginConfigurations() {}
+
+  public <T extends S3ClientBuilder> void configureSdkPlugin(T builder) {
```

Comment:
nit: I think it is more consistent to call it `configureS3ClientBuilder`?

Reply:
Done